openlibrary/plugins/openlibrary/code.py

```diff
@@ -427,6 +427,13 @@ def GET(self, key, value, suffix=''):
                 result = web.ctx.site.things(q)
                 if result:
                     return web.found(result[0] + ext)
+
+            # Perform import, if possible
+            from openlibrary.plugins.importapi.code import ia_importapi
+            from openlibrary import accounts
+            with accounts.RunAs('ImportBot'):
+                book = ia_importapi.ia_import(value, require_marc=True)
```

Comment:
This is kind of sneaking in past the API interface (bypassing auth?) to add a book, which worries me a bit as it's adding another path. 

Also I don't think the result here is a `book`, it's JSON suitable for an API response listing what Edition, Work and/or Author were matched / modified/ created

Oh, it looks like `book` isn't used - what happens when this code is triggered?

Does /books/ia: redirect? If so there might be a race condition for whether a potentially just created record is picked up. I'm not sure what happens here.

Reply:
1. we removed unused `book = `
2. https://dev.openlibrary.org/books/ia:mummycode0000hamm/The_mummy_code does **not** redirect. Eventually it should if we want to deprecate the "fake" book pages.
3. auth; this is the same thing we're doing for /isbn endpoint. It does possibly enable someone from the community to import from IA, however we tested that it's rejecting prohibited cases; e.g. https://dev.openlibrary.org/ia/sociobiologythenewsynthesis2000?debug=true